server/src/sql-tools/schema-diff.spec.ts

```diff
@@ -0,0 +1,274 @@
+import { schemaDiff } from 'src/sql-tools/schema-diff';
+import {
+  DatabaseColumn,
+  DatabaseColumnType,
+  DatabaseConstraint,
+  DatabaseIndex,
+  DatabaseSchema,
+  DatabaseTable,
+} from 'src/sql-tools/types';
+import { describe, expect, it } from 'vitest';
+
+const fromColumn = (column: Partial<Omit<DatabaseColumn, 'tableName'>>): DatabaseSchema => {
+  const tableName = 'Table1';
+
+  return {
+    name: 'public',
+    tables: [
+      {
+        name: tableName,
+        columns: [
+          {
+            name: 'Column1',
+            synchronize: true,
+            isArray: false,
+            type: 'character varying',
+            nullable: false,
+            ...column,
+            tableName,
+          },
+        ],
+        indexes: [],
+        constraints: [],
+        synchronize: true,
+      },
+    ],
+    warnings: [],
+  };
+};
+
+const newSchema = (schema: {
+  name?: string;
+  tables: Array<{
+    name: string;
+    columns?: Array<{
+      name: string;
+      type?: DatabaseColumnType;
+      nullable?: boolean;
+      isArray?: boolean;
+    }>;
+    indexes?: DatabaseIndex[];
+    constraints?: DatabaseConstraint[];
+  }>;
+}): DatabaseSchema => {
+  const tables: DatabaseTable[] = [];
+
+  for (const table of schema.tables || []) {
+    const tableName = table.name;
+    const columns: DatabaseColumn[] = [];
+
+    for (const column of table.columns || []) {
+      const columnName = column.name;
+
+      columns.push({
+        tableName,
+        name: columnName,
+        type: column.type || 'character varying',
+        isArray: column.isArray ?? false,
+        nullable: column.nullable ?? false,
```

Comment:
```suggestion
        isArray: !!column.isArray,
        nullable: !!column.nullable,
```

Reply:
I feel like this reads different. The one I use reads "use the value, or default it to false". The one you wrote reads "convert it to a boolean". I may want to have the default be something different in the future and then I'd have to change it back to my version to then change it to `?? true`.